client/src/components/TimeTable.jsx

```diff
@@ -0,0 +1,201 @@
+                    Hours:"4",
+                    Section: "L1"
+                },
+                {
+                    Name: "Software Engineering",
+                    ID: "IS F341",
+                    Days: "Monday Wednesday Friday",
+                    Hours:"7",
+                    Section: "L1"
+                },
+                {
+                    Name: "EVS",
+                    ID: "BITS F225",
+                    Days: "Tuesday Thursday Saturday",
+                    Hours:"2",
+                    Section: "L1"
+                },
+                {
+                    Name: "Post Colonial Literature",
+                    ID: "HSS F340",
+                    Days: "Tuesday Thursday Saturday",
+                    Hours:"3",
+                    Section: "L1"
+                },
+                {
+                    Name: "MicroProcessors",
+                    ID: "CS F241",
+                    Days: "Tuesday",
+                    Hours:"1",
+                    Section: "T11"
+                },
+                {
+                    Name: "DSA",
+                    ID: "CS F211",
+                    Days: "Thursday",
+                    Hours:"1",
+                    Section: "T3"
+                },
+                
+                {
+                    Name: "DSA",
+                    ID: "CS F211",
+                    Days: "Tuesday Thursday Saturday",
+                    Hours:"5 6 7 8",
+                    Section: "L1"
+                },
+                
+            ]
+        }
+        this.populateTimetable.bind(this);
+        this.gridArray = this.populateTimetable(this.state.courses); 
+    }
+    populateTimetable(coursesAdded)
+    {
+        var gridList = [
+            ["Time",1,2,3,4,5,6,7,8,9,10],
+            ["Monday",0,0,0,0,0,0,0,0,0,0],
+            ["Tuesday",0,0,0,0,0,0,0,0,0,0],
+            ["Wednesday",0,0,0,0,0,0,0,0,0,0],
+            ["Thursday",0,0,0,0,0,0,0,0,0,0],
+            ["Friday",0,0,0,0,0,0,0,0,0,0],
+            ["Saturday",0,0,0,0,0,0,0,0,0,0]
+        ];
+       
+        var Map = {};
+         Map['Monday'] = 1;
+         //Map[1] = 'Monday';
+         Map['Tuesday'] = 2;
+         //Map[2] = 'Tuesday';
+         Map['Wednesday'] = 3;
+         //Map[3] = 'Wednesday';
+         Map['Thursday'] = 4;
+         //Map[4] = 'Thursday';
+         Map['Friday'] = 5;
+         //Map[5] = 'Friday';
+         Map['Saturday'] = 6;
+         //Map[6] = 'Saturday';
+            
+            //let each course have a day list, and a hours list, both space separated strings
+            for(var course of coursesAdded)
+            {    
+                
+                var listOfDays = Array.from(course.Days.split(' '));
+                var listOfHours = (Array.from(course.Hours.split(' ')));
+                for(var i = 0; i< listOfHours.length; i++)
+                {
+                    listOfHours[i] = +listOfHours[i];
```

Comment:
this works?

Reply:
Yes. And parseInt doesn't for some reason :/

Reply:
Changed to include parseInt , have a look. I wasn't able to do it without a regular loop tho